### PR TITLE
feat: update template with useful examples [Part 1 of 2]

### PR DESCRIPTION
**Pull request by alanthai**
(Note - I need to update the templates to reflect the new `container` changes @vividviolet is working on https://github.com/Shopify/app-extension-libs/pull/757)

Feature: https://github.com/Shopify/app-extension-libs/issues/739
Fixes: https://github.com/Shopify/app-extension-libs/issues/758

Testing:
Clear your localStorage before testing.

generate for all template types (VanillaJS, TypeScript, React, React-TypeScript) and run to see it all works.

Create
Screen Shot 2020-06-17 at 4 49 41 PM

Edit (same as Create)
Screen Shot 2020-06-17 at 4 49 48 PM

Remove
Screen Shot 2020-06-17 at 4 58 28 PM

Add
Screen Shot 2020-06-17 at 4 50 23 PM

### CODE REVIEW

File: host/Host.tsx [hunk 1]
```diff
@@ -32,7 +31,7 @@ function HostWrapper() {
 
   return (
     <AppProvider theme={theme} i18n={{}}>
-      <Host script={reactThirdPartyWorker.url} components={components} />
+      <Host script={reactThirdPartyWorker.url} />
```
Comment (alanthai) [inline host/Host.tsx:34]: This was allowing us to use `Card` (not allowed in SubscriptionManagement)

File: scripts/generate/index.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ See README.md for instructions.
   console.log('✅ You selected:', choiceMap[template]);
 
   try {
-    generateSrc(extensionType, template as Template);
+    generateSrc(extensionType, choiceMap[template] as Template);
```
Comment (alanthai) [inline scripts/generate/index.ts:51]: Everything up until now was giving us Vanilla JS.

Comment (Fionoble) [reply scripts/generate/index.ts:51]: oops lol

**Comment by Fionoble** [comment]
Also, the yarn.lock should be updated with the latest argo version
I had to blow it away in order to get it to install the latest one

File: scripts/generate/templates/SUBSCRIPTION_MANAGEMENT/vanilla.ts.template [hunk 3]
```diff
@@ -1,75 +1,234 @@
...
+          ? selectedPlans.concat(plan.id)
+          : selectedPlans.filter((id) => id !== plan.id);
+
+        checkbox.updateProps({
+          checked: selectedPlans.includes(plan.id),
+        });
+      },
+    });
+
+    stack.appendChild(checkbox);
```
Comment (vividviolet) [inline scripts/generate/templates/SUBSCRIPTION_MANAGEMENT/vanilla.ts.template:60]: TS shows an error for this and a bunch of other `appendChild` calls for some reason

image

Comment (alanthai) [reply scripts/generate/templates/SUBSCRIPTION_MANAGEMENT/vanilla.ts.template:55]: I need @henrytao-me 's PR for it to be fixed.

https://github.com/Shopify/app-extension-libs/pull/755#discussion_r441795536

**Comment by alanthai** [comment]
@Fionoble @vividviolet could you try clearing localStorage? I updated it in the comments.

**Comment by Fionoble** [comment]
I'd like to get @iainmcampbell 's feedback on this before shipping though

**Pull request merged**